ipaclient/install/client.py

```diff
@@ -1672,12 +1672,21 @@ def client_dns(server, hostname, options):
 
     # Setup DNS over TLS
     if options.dns_over_tls:
+        statestore.backup_state("dns_over_tls", "enabled", True)
+        save_state(services.knownservices["unbound"], statestore)
         # setup and enable Unbound as resolver
         server_ip = str(list(dnsutil.resolve_ip_addresses(server))[0])
         forward_addr = "forward-addr: %s#%s" % (server_ip, server)
         # module_config_iterator is commented out if DNSSEC validation is
         # not disabled.
         module_config_iterator = '' if options.no_dnssec_validation else '# '
+        # backup and remove all previous Unbound configuration
+        for filename in os.listdir(paths.UNBOUND_CONFIG_DIR):
+            filepath = os.path.join(paths.UNBOUND_CONFIG_DIR, filename)
+            if filepath == paths.UNBOUND_CONF:
+                continue
+            fstore.backup_file(filepath)
```

Comment:
Did you mean statestore.backup_file(filepath) rather than fstore.backup_file(filepath)?

Reply:
No, statestore does not handle file backup. It's failing because I was using a variable declared later, I've fixed it.